### DISCUSSION TITLE
zoom in & out features are strongly needed...

### DISCUSSION THREAD

**Discussion by int22h** [ihhub/fheroes2#8065] [Ideas]
specially when i'm tracking around the land in the game on ANDROID.  objects are too small to touch on.
just my hope.   it will be great if i can zoom the land.

**Comment by oleg-derevenetz** [comment]
"Objects are too small to touch on"? Which in-game resolution you use? On handheld devices there is usually Nx480 resolution used, where N is calculated depending on screen aspect ratio, which is in fact the minimum resolution that will cover the entire screen. For example, on my phone it's 1120x480. This resolution has no issues with object sizes, unless the screen itself is too small. Using higher resolutions is not recommended. And playing on devices with too small (physically) screen will be inconvenient in any case. It is recommended to use either modern smartphone with relatively large screen or tablet (even ancient ones with Android 5.1 will go, because fheroes2 is undemanding to hardware resources).

**Reply by int22h** [reply]
Thank you so much for the comment :)

I'm not just complaining that fheroes2 is uncomfortable to play. Actually, I found the solution to the issue of right-clicking on objects yesterday :D

As an example, however, consider the game OpenTTD. The original Transport Tycoon Deluxe only had three zoom levels.
Although it is limited... Still, with OpenTTD, you can now look a little closer and a little farther than before.
Was that just a feature added because it was inconvenient to see or play?

I think anyone can have the desire or romance to see the world in the games we love a little closer or further away.

And, this is just my personal hope :)
Also, I am playing on the tablet. There is no problem at all, now.

**Reply by oleg-derevenetz** [reply]
>As an example, however, consider the game OpenTTD. The original Transport Tycoon Deluxe only had three zoom levels.
Although it is limited... Still, with OpenTTD, you can now look a little closer and a little farther than before.

In the original TTD, this mechanics was already there, just a bit more limited. In the original HoMM2 there was nothing like that at all. To introduce this, there will be a need to create the corresponding mechanics (not just for touch, but also for devices with mouse, because this should work everywhere, otherwise, the players on non-touch platforms will start whining) which will certainly require changes to the original UI, new home-grown controls, etc., which we would like to avoid as much as possible.

**Comment by Branikolog** [comment]
Here's a collage I made for a fun. I zoomed the adventure map precisely for x2.

![Adventure map x1](https://github.com/ihhub/fheroes2/assets/55348946/ce997478-d9cf-4203-8ab6-8f648d3ed4d7)
![Adventure map x2](https://github.com/ihhub/fheroes2/assets/55348946/dddb7e2a-5a1b-4b5e-befa-4fa1a7327d85)
